docs/Testing.md

```diff
@@ -43,6 +44,12 @@ uses an iOS 10.3 simulator configuration named "iPhone 5" for 32-bit testing.
 
 3.  Append `--ios` to the `utils/build-script` command line (see below).
 
+macOS 11 (Big Sur) drops support for the iOS 10.3 simulator. If you want to
+reproduce a 32-bit iOS simulator failure, you can instead try using the 32-bit
+watchOS simulator using `build-script`'s `--watchos` flag. If you are
+additionally using the `--swift-darwin-supported-archs` flag, include
+`i386` in the list of architectures.
+
```

Comment:
There's another issue (#33450) — an Intel-based Mac is required, because Macs with Apple silicon don't support 32-bit processes, according to the [Xcode 12 Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-12-beta-release-notes/).

> **Deprecations**
> 
> * When running in macOS 11, Simulator supports iOS 11.4 or later. (59938106)
> 
> * Simulators for watchOS 6 or earlier require 32-bit processes that are not supported on Macs with Apple silicon. (66352760)
> 
> **New Features**
> 
> * Simulator now supports 64-bit processes on watchOS 7. Make sure your project uses $(STANDARD_ARCHS) to build for x86_64 on Intel Macs, and for arm64 on Macs with Apple silicon. This configuration allows you to check your code is 64-bit clean, and makes it possible to simulate a watchOS device on Apple silicon, which doesn’t support 32-bit processes. (66352876)
> 
> * Simulator now supports 64-bit and 32-bit processes for watchOS 7. To verify watchOS projects are 64-bit clean in Simulator, make sure ARCHS is set to its default value. (54654060)

Reply:
I think `get_simulator_command` in [test/lit.cfg](https://github.com/apple/swift/blob/789659e0d770a10421573f90785385368e0455ff/test/lit.cfg#L78-L99) also needs to be updated.

Reply:
Thanks. I haven't looked too deeply into how the runtime detection works in `test/lit.cfg`, we should probably raise an error in certain cases. Will get to this next week.